### DISCUSSION TITLE
When writing Markdown, how do I use roman numerals in deeply nested lists?

### DISCUSSION THREAD

**Discussion by Jayman2000** [community/community#124850] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

[I’m currently working on improving a project’s installation instructions.](https://github.com/DescentDevelopers/Descent3/pull/297) The instructions are rather complicated, so I want to organize them into nested lists. I want those nested lists to use different numbering schemes like this:

```text
1. Step
    • Small note.
    • Very large note:
        a. Step
            i. Step
                1. Step
```

Unfortunately, I’m having trouble getting the roman numerals to work.

#### Attempt 1

First, I tried writing Markdown like this:

```Markdown
1. Step
    - Small note.
    - Very large note:
        1. Step
            1. Step
                1. Step
```

Here’s how GitHub renders that Markdown:

1. Step
    - Small note.
    - Very large note:
        1. Step
            1. Step
                1. Step

The problem here is that all of the sublists inside the very large note use letters. None of them use Roman or Arabic numerals.

#### Attempt 2

After that, I tried embedding raw HTML in the Markdown:

```Markdown
1. Step
    - Small note.
    - Very large note:

        <ol type="a">
            <li>
                <p>Step</p>
                <ol type="i">
                    <li>
                        <p>Step</p>
                        <ol type="1">
                            <li><p>Step</p></li>
                        </ol>
                    </li>
                </ol>
            </li>
        </ol>
```

Here’s how GitHub renders that Markdown:

1. Step
    - Small note.
    - Very large note:

        <ol type="a">
            <li>
                <p>Step</p>
                <ol type="i">
                    <li>
                        <p>Step</p>
                        <ol type="1">
                            <li><p>Step</p></li>
                        </ol>
                    </li>
                </ol>
            </li>
        </ol>

The problem here is that one of the lists uses letters even though I told it to use Roman numerals. When I inspect the generated HTML, I can see that the `ol` elements do indeed have `type` attributes. For whatever reason, setting `type` to `i` doesn’t work for deeply nested lists.

#### Attempt 3

After that, I tried using the `style` attribute:

```Markdown
1. Step
    - Small note.
    - Very large note:

        <ol style="list-style-type: lower-alpha;">
            <li>
                <p>Step</p>
                <ol style="list-style-type: lower-roman;">
                    <li>
                        <p>Step</p>
                        <ol style="list-style-type: decimal;">
                            <li><p>Step</p></li>
                        </ol>
                    </li>
                </ol>
            </li>
        </ol>
```

Here’s how GitHub renders that Markdown:

1. Step
    - Small note.
    - Very large note:

        <ol style="list-style-type: lower-alpha;">
            <li>
                <p>Step</p>
                <ol style="list-style-type: lower-roman;">
                    <li>
                        <p>Step</p>
                        <ol style="list-style-type: decimal;">
                            <li><p>Step</p></li>
                        </ol>
                    </li>
                </ol>
            </li>
        </ol>

The problem here is that all of the sublists inside the very large note use letters again. When I inspect the generated HTML, I can see that the `ol` elements don’t have `style` attributes because [GitHub explicitly removes them](https://github.com/github/markup/blob/a849a12ac5100d7d7acba30834eab73d0cf61983/README.md#github-markup).

#### Conclusion

At this point, I’m out of ideas. How do I use Roman numerals in deeply embedded lists?

**Comment by davevad93** [comment]
Hi @Jayman2000 ,

I don't know if there's a practical way to do it, one solution I can think of would be to manually write the Roman numerals, I know it's a rather unorthodox way but it's the only solution I can think of to be honest:

1. Step
   - Small note.
   - Very large note:
       1. Step
           i. nested roman numeral
           ii. nested roman numeral
           iii. nested roman numeral
       2. Step
           i. nested roman numeral
           ii. nested roman numeral
           iii. nested roman numeral
           iv. nested roman numeral
2. Step
   - Small note.
   - Very large note:
       1. Step
           i. nested roman numeral
           ii. nested roman numeral
           iii. nested roman numeral
           iv. nested roman numeral
           v. nested roman numeral
       2. Step
           i. nested roman numeral
           ii. nested roman numeral
           iii. nested roman numeral

```Markdown
1. Step
   - Small note.
   - Very large note:
       1. Step
           i. nested roman numeral
           ii. nested roman numeral
           iii. nested roman numeral
       2. Step
           i. nested roman numeral
           ii. nested roman numeral
           iii. nested roman numeral
           iv. nested roman numeral
2. Step
   - Small note.
   - Very large note:
       1. Step
           i. nested roman numeral
           ii. nested roman numeral
           iii. nested roman numeral
           iv. nested roman numeral
           v. nested roman numeral
       2. Step
           i. nested roman numeral
           ii. nested roman numeral
           iii. nested roman numeral
```

In an `HTML` file this works but as you can see `Markdown` doesn't natively support list styles like `HTML` does:

<ol>
    <li>Step
        <ul>
            <li>Small note.</li>
            <li>Very large note:
                <ol>
                    <li>Step
                        <ol type="i">
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                        </ol>
                    </li>
                    <li>Step
                        <ol type="i">
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                        </ol>
                    </li>
                </ol>
            </li>
        </ul>
    </li>
    <li>Step
        <ul>
            <li>Small note.</li>
            <li>Very large note:
                <ol>
                    <li>Step
                        <ol type="i">
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                        </ol>
                    </li>
                    <li>Step
                        <ol type="i">
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                        </ol>
                    </li>
                </ol>
            </li>
        </ul>
    </li>
</ol>

```HTML
<ol>
    <li>Step
        <ul>
            <li>Small note.</li>
            <li>Very large note:
                <ol>
                    <li>Step
                        <ol type="i">
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                        </ol>
                    </li>
                    <li>Step
                        <ol type="i">
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                        </ol>
                    </li>
                </ol>
            </li>
        </ul>
    </li>
    <li>Step
        <ul>
            <li>Small note.</li>
            <li>Very large note:
                <ol>
                    <li>Step
                        <ol type="i">
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                        </ol>
                    </li>
                    <li>Step
                        <ol type="i">
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                            <li>nested roman numeral</li>
                        </ol>
                    </li>
                </ol>
            </li>
        </ul>
    </li>
</ol>
```

**Reply by Jayman2000** [reply]
> I don't know if there's a practical way to do it, one solution I can think of would be to manually write the Roman numerals, I know it's a rather unorthodox way but it's the only solution I can think of to be honest:

That works surprisingly well. Thank you!

**Reply by davevad93** [reply]
Hey @Jayman2000 , you're welcome I'm glad you found it helpful.
